lntest/itest/assertions.go

```diff
@@ -623,6 +623,121 @@ func txStr(chanPoint *lnrpc.ChannelPoint) string {
 	return cp.String()
 }
 
+// waitForChannelUpdate waits for a node to receive the expected channel
+// updates.
+func waitForChannelUpdate(t *harnessTest, subscription graphSubscription,
+	expUpdates []expectedChanUpdate) {
+
+	// Create an array indicating which expected channel updates we have
+	// received.
+	found := make([]bool, len(expUpdates))
+out:
+	for {
+		select {
+		case graphUpdate := <-subscription.updateChan:
+			for _, update := range graphUpdate.ChannelUpdates {
+				require.NotZerof(
+					t.t, len(expUpdates),
+					"received unexpected channel "+
+						"update from %v for channel %v",
+					update.AdvertisingNode,
+					update.ChanId,
+				)
+
+				// For each expected update, check if it matches
+				// the update we just received.
+				for i, exp := range expUpdates {
+					fundingTxStr := txStr(update.ChanPoint)
+					if fundingTxStr != txStr(exp.chanPoint) {
+						continue
+					}
+
+					if update.AdvertisingNode !=
+						exp.advertisingNode {
+						continue
+					}
+
+					err := checkChannelPolicy(
+						update.RoutingPolicy,
+						exp.expectedPolicy,
+					)
+					if err != nil {
+						continue
+					}
+
+					// We got a policy update that matched
+					// the values and channel point of what
+					// we expected, mark it as found.
+					found[i] = true
+
+					// If we have no more channel updates
+					// we are waiting for, break out of the
+					// loop.
+					rem := 0
+					for _, f := range found {
+						if !f {
+							rem++
+						}
+					}
+
+					if rem == 0 {
+						break out
+					}
+
+					// Since we found a match among the
+					// expected updates, break out of the
+					// inner loop.
+					break
+				}
+			}
+		case err := <-subscription.errChan:
+			t.Fatalf("unable to recv graph update: %v", err)
+		case <-time.After(defaultTimeout):
```

Comment:
We may need to put this `time.After` outside the for loop, otherwise we will wait `defaultTimeout` each time the loop circles.